service/src/main/java/org/broadinstitute/listener/relay/inspectors/TokenChecker.java

```diff
@@ -41,12 +41,27 @@ public OauthInfo getOauthInfo(String token) throws IOException, InterruptedExcep
     }
   }
 
+  public Optional<String> getClaim(String token, String claim) {
+    try {
+      var decoded = JWT.decode(token);
```

Comment:
Er, I guess google tokens may not have all the claims we need but it should have `sub` probably..

Reply:
Hmm can a Google token be used against Relay? Or I guess we should be detecting when it is, as another an unexpected scenario (and literally "Use of service accounts for interactive log on" in the compliance checklist)

Reply:
And more specifically, we could detect that based on `.gserviceaccount.com` suffix in the `email` claim.

Reply:
I mean technically any valid Terra token could be used against Relay. Possibilities:

1. Google user token -- e.g. if someone does `gcloud auth print-access-token` and makes a request to Relay. This is probably not common, I can't think of a Terra use case that would result in this.
2. Google service account token -- e.g. Terra control plane service calling relay. This happens during Leo polling, potentially other places.
3. B2C user token -- user opening an app in Terra UI
4. UAMI token -- data plane apps talking to each other, example CBAS and WDS

Reply:
Good catch @rtitle , I'll look into reusing the decoded + google token fallback we're using the sam stuff. So, it sounds like we'll want to emit the email claim (if present), will that suffice for the pet detection @dinvlad ?

Reply:
I think it'd be fine to log the email tbh, and we can always improve that later if there's a privacy concern. It'd be easier to review the alerts this way, although it's not strictly required for non-control plane SA detection (since we could log just the fact that the host name part of the email starts with the Google project ID in the control plane).